file_changed_command_runner/plugin.py

```diff
@@ -0,0 +1,38 @@
+""" Atakama plugin: FileChangedCommandRunner """
+
+import logging
+import shlex
+import subprocess
+from typing import Dict
+from atakama import FileChangedPlugin
+
+from .callback_server import CallbackServer
+
+
+log = logging.getLogger(__name__)
+
+
+class FileChangedCommandRunner(FileChangedPlugin):
+    """Atakama plugin: FileChangedCommandRunner"""
+
+    def __init__(self, args: Dict[str, str]):
+        super().__init__(args)
+        self._cmd: str = args["cmd"]
+        self._timeout: float = args.get("timeout", 10)
+        self._callback_server: CallbackServer = CallbackServer()
+
+    @staticmethod
+    def name():
+        return "file-changed-command-runner"
+
+    def file_changed(self, full_path: str) -> None:
+        with self._callback_server as cb:
+            cmd = f"{shlex.quote(self._cmd)} {shlex.quote(full_path)} {shlex.quote(cb.url)}"
+            log.debug("about to run command: %s", cmd)
+            with subprocess.Popen(cmd, shell=True) as proc:
```

Comment:
Can this be changed to not use `shell=True`? Things look to be correctly quoted, but avoiding the [security problems with `shell=True`](https://docs.python.org/3/library/subprocess.html#security-considerations) altogether seems preferable.

Reply:
this was a holdover from one of the early sample plugins. no need for it now, removed.